torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -435,6 +435,7 @@ class TORCH_API ProcessGroupNCCL : public Backend {
     int64_t split_color{0};
     std::vector<uint64_t> global_ranks_in_group;
     std::string group_name;
+    std::string group_desc;
```

Comment:
Like other comments above, move this to ProcessGroup.hpp?

Reply:
Since we only pass in group_desc in processGroupNCCL when creating it, I think leave it here would be better together with group_name? For other backends the group_name/desc will be set at python layer after pg creation.

Reply:
Same as above, no need to follow something that's wrong.

Reply:
We need a way to pass in group name/desc for a backend when creating it. Setting through ProcessGroup cannot allow us accessing these information in the backend init function.

Reply:
Not only printing a log, but also pass to profiler, and flight record to know pg set up.

Reply:
If the information is in the `Backend` class, which is the base for `ProcessGroupNCCL`, profiler & FR should be able to access it.